core/src/data.ts

```diff
@@ -77,9 +81,10 @@ export function converterFromSpec(spec: ConverterSpec) {
  * `NoteSequence` objects.
  */
 export abstract class DataConverter {
-  abstract numSteps: number;  // Total length of sequences.
-  abstract numSegments: number;  // Number of steps for conductor.
-  abstract readonly NUM_SPLITS: number;  // Number of conductor splits.
+  abstract readonly numSteps: number;  // Total length of sequences.
+  abstract readonly numSegments: number;  // Number of steps for conductor.
+  abstract readonly depth: number;  // Size of final output dimension.
+  abstract readonly NUM_SPLITS: number;  // Const number of conductor splits.
```

Comment:
now that the other properties are also readonly, should they either all be uppercase or should NUM_SPLITS be lowercase?

Reply:
No, NUM_SPLITS is still something different. It really should be const but const isn't allowed in the abstract definition...